src/cargo/util/flock.rs

```diff
@@ -430,30 +430,48 @@ fn is_on_nfs_mount(_path: &Path) -> bool {
     false
 }
 
+#[cfg(all(unix, not(target_os = "solaris")))]
```

Comment:
Sorry for being picky. Feel like this can move one level down into the `#[cfg(unix)] mod sys` so we can remove `all(unix,` part from cfg?